src/FuelClient.cc

```diff
@@ -1230,6 +1239,12 @@ Result FuelClient::CachedModelFile(const common::URI &_fileUrl,
   // Check if file exists
   filePath = common::joinPaths(modelPath, filePath);
 
+  std::vector<std::string> tokens = ignition::common::split(filePath, "/");
```

Comment:
This logic is being added to `ParseWorldFileUrl` and to `CachedModelFile`. I think we should be consistent between world and file, so either add it to both `Parse*FleUrl` functions or to both `Cached*File`s.